src/algorithms/kss.jl

```diff
@@ -59,20 +59,20 @@ and subspace basis matrices `U[1],...,U[K]`.
 - `maxiters::Integer = 100`: maximum number of iterations
 - `rng::AbstractRNG = default_rng()`: random number generator
     (used when reinitializing the subspace for an empty cluster)
-- `Uinit::AbstractVector{<:AbstractMatrix{<:AbstractFloat}}
+- `Uinit::AbstractVector{<:AbstractMatrix{T}}
     = [randsubspace(rng, size(X, 1), di) for di in d]`:
     vector of `K` initial subspace basis matrices to use
     (each `Uinit[k]` should be `D×d[k]`)
 
 See also [`KSSResult`](@ref).
 """
 function kss(
-    X::AbstractMatrix{<:Real},
+    X::AbstractMatrix{T} where T<:Number,
     d::AbstractVector{<:Integer};
     maxiters::Integer = 100,
     rng::AbstractRNG = default_rng(),
-    Uinit::AbstractVector{<:AbstractMatrix{<:AbstractFloat}} = [
-        randsubspace(rng, size(X, 1), di) for di in d
+    Uinit::AbstractVector{<:AbstractMatrix{<:Number}} = [
+        randsubspace(rng, eltype(X), size(X, 1), di) for di in d
```

Comment:
Let's ensure that `Uinit` is a floating-point type like so:

```suggestion
        randsubspace(rng, float(eltype(X)), size(X, 1), di) for di in d
```